kubernetes-deployment/massdriver.yaml

```diff
@@ -18,61 +18,98 @@ params:
   properties:
     namespace:
       title: Kubernetes Namespace
-      type: string
+      description: Application will be deployed into this namespace. If the namespace doesn't exist, it will be created.
+      $ref: https://raw.githubusercontent.com/massdriver-cloud/artifact-definitions/main/definitions/types/k8s-resource-name.json
       default: default
     image:
       title: Container Image
       type: object
+      required:
+      - repository
+      - tag
       properties:
         repository:
+          $md.immutable: true
```

Comment:
Making the container immutable, so its settable in the UI, but can't be accidentally swapped to a different image.

Reply:
👍🏼